extend.sh

```diff
@@ -271,9 +330,18 @@ function resize_fs(){
     fi
 }
 
-function shrink_partition(){
-    local device=$DEVICE_NAME$SUCCESSIVE_PARTITION_NUMBER
-    device_type=$(get_device_type "$device")
+function shrink_partition() {
+    new_end_partition_size_in_bytes=$(calculate_new_end_partition_size_in_bytes "$1")
```

Comment:
Consider assigning `$1` to a local variable to make the code easier to read

Reply:
Done